raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -190,18 +190,18 @@ contract TokenNetwork is Utils {
         Channel storage channel = channels[channel_identifier];
         Participant storage participant_state = channel.participants[participant];
 
-        require(participant_state.deposit < total_deposit);
-
         // Calculate the actual amount of tokens that will be transferred
         added_deposit = total_deposit - participant_state.deposit;
 
         // Update the participant's channel deposit
         participant_state.deposit += added_deposit;
 
+        emit ChannelNewDeposit(channel_identifier, participant, participant_state.deposit);
+
         // Do the transfer
-        require(token.transferFrom(msg.sender, address(this), added_deposit));
+        assert(token.transferFrom(msg.sender, address(this), added_deposit));
 
-        emit ChannelNewDeposit(channel_identifier, participant, participant_state.deposit);
+        assert(participant_state.deposit >= added_deposit);
```

Comment:
also a require() here, not an assert. In general assert should be for something that should never ever happen under any circumstances, i.e.: internal errors.